### PR TITLE
chore: drop support for go1.20 

### PR DESCRIPTION
**Pull request by sukunrt**

### CODE REVIEW

**Review by Stebalien — APPROVED** [review]
From what I can tell, go 1.20 still "works", it'll just use a slower sha implementation. Which is fine, IMO.

File: core/crypto/ecdsa.go [hunk 1]
```diff
@@ -10,9 +10,10 @@ import (
 	"io"
 	"math/big"
```
Comment (Jorropo) [inline core/crypto/ecdsa.go:12]:
```suggestion
```

File: core/crypto/rsa_go.go [hunk 2]
```diff
@@ -8,9 +8,10 @@ import (
 	"errors"
 	"io"
```
Comment (Jorropo) [inline core/crypto/rsa_go.go:10]:
```suggestion
```

File: p2p/net/swarm/swarm_addr_test.go [hunk 3]
```diff
@@ -5,11 +5,12 @@ import (
 	"fmt"
 	"testing"
```
Comment (Jorropo) [inline p2p/net/swarm/swarm_addr_test.go:7]:
```suggestion
```

File: p2p/security/noise/handshake.go [hunk 4]
```diff
@@ -10,10 +10,11 @@ import (
 	"runtime/debug"
 	"time"
```
Comment (Jorropo) [inline p2p/security/noise/handshake.go:12]:
```suggestion
```

**Review by Jorropo — APPROVED** [review]

**Pull request merged**